### PR TITLE
Added Wilcoxon, Tarone-Ware and Peto log-rank alternative tests

### PR DESCRIPTION
**Pull request by sean-reed**
See issue #452

### CODE REVIEW

File: lifelines/tests/test_statistics.py [hunk 1]
```diff
@@ -103,6 +103,48 @@ def test_rank_test_output_against_R_with_censorship():
     assert abs(result.test_statistic - r_stat) < 10e-6
 
 
+def test_wilcoxon_weighted_logrank_on_leukemia_dataset():
```
Comment (CamDavidsonPilon) [inline lifelines/tests/test_statistics.py:106]: 👍 nice

File: lifelines/statistics.py [hunk 2]
```diff
@@ -646,6 +675,18 @@ def multivariate_logrank_test(
     t_0: float, optional (default=-1)
         the period under observation, -1 for all time.
 
+    weightings: str, optional
+        apply a weighted logrank test: options are "w" for Wilcoxon (also known as Breslow), "tw"
```
Comment (CamDavidsonPilon) [inline lifelines/statistics.py:679]: I feel these should be more explicit. Ex: `weightings="wilcoxon"` vs `weights="w"`. The latter feels like an R thing, and the former a Python thing (at least, how I want Python to be). Thoughts?

**Comment by CamDavidsonPilon** [comment]
Love it, great addition. Can you comment about _why_ one would want to use different weightings?

File: lifelines/statistics.py (hunk 2 above)
Comment (sean-reed) [reply lifelines/statistics.py:679]: Yes, that seems better.

**Comment by sean-reed** [comment]
The weighted tests put emphasis on the beginning of the survival curve (e.g. Wilcoxon where weight is based on remaining number at risk), increasing the power of the test when it's expected the difference in survival is early on,  whereas the standard log-rank test has maximum power for proportional hazards. For example, certain drugs reduce death rates early on (e.g. first 10 years over 20 year follow up) but have less effect on the death rate for those that survive beyond a certain point (e.g. between years 10 and 20). Another example might be an online store testing whether including a discount voucher with a customer's order influences the time to them placing their next order. You might expect it to influence the rate of re-ordering for the first few weeks after receiving the voucher but have a diminishing influence later on, so a weighted test may be more appropriate. Should add that the choice of whether to weight the test should be made in advance to avoid introducing bias. I can add something to the docs on this at some point.

**Comment by CamDavidsonPilon** [comment]
>  I can add something to the docs on this at some point.

What you wrote is great 👍

**Comment by CamDavidsonPilon** [comment]
PR is great - did you want to add documentation to `/docs` or do you want me to add later?

**Comment by sean-reed** [comment]
Thanks, leave it to me I'll add it.

**Comment by sean-reed** [comment]
Added some docs and also the Fleming-Harrington test (https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6047314/#bib4) that allows testing for late differences in survival.

File: docs/Examples.rst [hunk 3]
```diff
@@ -93,6 +93,56 @@ hypothesis that all the populations have the same "death" generation process).
     ---
     """
 
+The logrank test statistic is calculated from the differences between the observed deaths for a group and expected
```
Comment (CamDavidsonPilon) [inline docs/Examples.rst:96]: Awesome!

**Pull request merged**